### ISSUE TITLE
[Ubuntu/arm][GCStress=0xC][ZapDisable] Regressions/coreclr/GitHub_12224/Test12224

### ISSUE THREAD

**Issue by echesakovMSFT** [dotnet/coreclr#19086]
This failed twice while testing PR #18784 in 
[Ubuntu arm Cross Checked gcstress0xc_zapdisable Build and Test](https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_tst_prtest/5/consoleText)
[Ubuntu arm Cross Checked gcstress0xc_zapdisable_heapverify1 Build and Test](https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_heapverify1_tst_prtest/4/consoleText)
```
FAILED   - Regressions/coreclr/GitHub_12224/Test12224/Test12224.sh
               BEGIN EXECUTION
               /ssd/j/workspace/dotnet_coreclr/master/jitstress/arm_cross_checked_ubuntu_gcstress0xc_zapdisable_tst_prtest/bin/tests/Linux.arm.Checked/Tests/Core_Root/corerun Test12224.exe
               Expected: 100
               Actual: 255
               END EXECUTION - FAILED
```

**Note:** when the issue is resolved remove the corresponding line in tests/testsFailing.arm.txt (https://github.com/dotnet/coreclr/pull/19087)

**Comment by AndyAyersMS** [comment]
I'll investigate.

**Comment by AndyAyersMS** [comment]
I normally cross-build from Linux x64 and realized this test wasn't getting built. Opened #22627.

With a workaround for that. I was able to repro the failure, if I added extra load to the machine.

This test has timing-dependent behavior. Not too surprising that it can exit with 255 under GC stress, especially when the machine is running other things concurrently.

https://github.com/dotnet/coreclr/blob/b929fcb92db1d64044bef9b1577e95e6e4da3f25/tests/src/Regressions/coreclr/GitHub_12224/test12224.cs#L37-L41

While we could increase the timeout, we still might hit occasional failures in stress. Suggest we mark this test as stress incompatible.
